launcher/main.go

```diff
@@ -176,6 +186,33 @@ func instanceId() string {
 	return enc.EncodeToString(sum[:])
 }
 
+func waitUntilFileAppears(filename string) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		panic(err)
+	}
+	defer watcher.Close()
+
+	if err := watcher.Add(filepath.Dir(filename)); err != nil {
+		panic(err)
+	}
+
+	if _, err := os.Stat(filename); err == nil {
+		return
+	}
```

Comment:
```suggestion
	if _, err := os.Stat(filename); err == nil {
		return
	}

	watcher, err := fsnotify.NewWatcher()
	if err != nil {
		panic(err)
	}
	defer watcher.Close()

	if err := watcher.Add(filepath.Dir(filename)); err != nil {
		panic(err)
	}
```
If we do the stat at the top, we can avoid creating the watcher.

Reply:
I would not change that, since this openes a race-condition, where we could miss the creation of the socket and therefore get into an infinite loop.
We first have to setup the watcher and only then check, if the socket was already created. When it is not created at that point, we should definitively see the creation in the watcher.